src/reynir_correct/pattern.py

```diff
@@ -178,15 +179,38 @@ def __init__(self, ann: List[Annotation], sent: Sentence) -> None:
     def get_wordform(self, lemma: str, cat: str, variants: Iterable[str]):
         """ Get correct wordform from BinPackage, 
         given a set of variants """
-
-        # Get rid of argument variants in verbs:
-        realvars = [v for v in variants if not v.isdigit() and v not in SKIPVARS]
+        realvars: Iterable[str]
+        if cat == "so":
+            # Get rid of argument variants in verbs:
+            realvars = frozenset([v for v in variants if not v.isdigit() and v not in SKIPVARS])
+            if not "lh" in realvars:
+                realvars = frozenset([v for v in realvars if v not in ALL_CASES])
+        else:
+            realvars= frozenset(variants)
```

Comment:
Need to run Black formatter on the code

Reply:
Fixed this manually. Am running into problems with Black where it deletes spaces in docstrings. Am looking into it.